config/elasticsearch_client.yml

```diff
@@ -0,0 +1,16 @@
+default: &DEFAULT
+  log: false
+  randomize_connections: true
+  reload_connections: true
+  reload_on_failure: true
+  retry_on_failure: 1
```

Comment:
why is retry_on_failure here 1 but below you set it to false. To be consistent can we use either true, false or 1,0 ?  or is that the number of retries? In that case should retry_on_failure be 0?

Reply:
The `retry_on_failure` value can either be a boolean or an integer indicating the # of times to retry. I am setting it to `1` to retry once in production, because I don't want it retrying the default 3 times (due to the potential for overloading the ES servers with retried queries when there are system issues, as we've seen with i14y). I've left the default value in test to mirror the prod values, but I set it to `false` in dev, because when developing I'd like a fast failure.

Reply:
would 0 give you the same result? It would be better to keep it consistent.

Reply:
The ES client code behaves differently depending on whether `retry_on_failure` is `false` vs. an integer (even if that integer is zero), so I'd like to keep it as-is.